src/PhpSpreadsheet/Style/ConditionalFormatting/CellMatcher.php

```diff
@@ -214,6 +215,15 @@ protected function processOperatorComparison(Conditional $conditional): bool
         return $this->evaluateExpression($expression);
     }
 
+    protected function processColorScale(Conditional $conditional): bool
+    {
+        if ($this->wrapCellValue() !== '' && is_float($this->wrapCellValue()) && $conditional->getColorScale() !== null && $conditional->getColorScale()->colorScaleReadyForUse()) {
```

Comment:
If I understand your code correctly your first 2 conditions conditions can be combined - if it's a float, then it's not equal to null-string. And I also believe the 3rd and 4th conditions can also be combined with a null-safe operator.
```php
if (is_float($this->wrapCellValue()) && $conditional->getColorScale?->colorScaleReadyForUse())
```
However, I'm not entirely sure about is_float. It seems that wrapCellValue might also return int. If you don't want that to follow the same code path as float, then leave is_float as-is; however, if you do want int and float to follow the same path, you might use is_numeric instead.